docs/en/reference/testing.rst

```diff
@@ -0,0 +1,74 @@
+Testing Guidelines
+===================
+
+In order to ensure high quality, all components the Doctrine DBAL library are extensively covered with tests.
+
+Having the code covered with tests and running all tests against each individual code change helps prevent
+breakages of the library logic when its code changes.
+
+Additionally, tests:
+
+1. Help maintainers understand what problem the given code change is trying to solve.
+2. Make sure that the problem being solved needs to be solved in the DBAL.
+3. Document the proper usage of the DBAL APIs.
+
+Requirements
+------------
+
+1. Each pull request that adds new or changes the existing logic must have tests.
+2. The test that covers certain logic must fail without the changes in the code it covers.
+
+Unit Tests
+----------
+
+Unit test are meant to cover the logic of a given unit (e.g. a class or a method) including the logic of its interaction
+with other units. In this case, the other units could be mocked.
+
+Unit tests are mostly welcomed for testing the logic that the DBAL itself defines (e.g. logging). They also could be
+used as a secondary tier of testing for the logic that requires integration testing.
```

Comment:
That last sentence could be clarified with an example.

Reply:
I moved this under the "Using Unit and Integration Tests Together" section. The wording is to be revisited.